engine/runtime/src/main/java/org/enso/interpreter/runtime/type/Types.java

```diff
@@ -112,24 +112,29 @@ public static void extractArguments(Object[] arguments) throws ArityException {
    * @return the string representation of object's type.
    */
   public static String getName(Object value) {
-    if (TypesGen.isLong(value) || TypesGen.isImplicitLong(value)) {
-      return "Number";
+    if (TypesGen.isLong(value)
+        || TypesGen.isDouble(value)
+        || TypesGen.isEnsoBigInteger(value)
+        || TypesGen.isImplicitLong(value)) {
+      return "Builtins.Main.Number";
```

Comment:
This doesn't seem like quite the correct thing, actually. Given we have methods applicable to `Integer` but not `Decimal`, you want to split these cases to return `Builtins.Main.Integer` for `isImplicitLong(...) || isEnsoBigInteger(...)` and return `Builtins.Main.Decimal` for `isDouble(...)`.